### PR TITLE
feat: add CentOS 8 support

### LINKED ISSUE #2033: Add CentOS 8 support
We'd like to add CentOS to the support matrix, making an exercise of how easy is to add a new OS/Arch.

cc/ @jlind23

### PR DESCRIPTION
**Pull request by mdelapenya**
- chore: rename platforms to include OS
- fix: include sles15 in Beats builds
- chore: expand coverage for SLES15
- chore: run k8s-autodiscover on Debian VMs
- chore: run helm test suite on Debian VMs
- chore: remove unused platforms
- feat: add centos8 support, running fleet test suite

## What does this PR do?
This PR adds CentOS 8 as a backend to run the tests, using the AMIs provided by the official Centos website: https://www.centos.org/download/aws-images/

We have also refactored the platforms YAML file to identify the platform with the OS name: i.e. `fleet_amd64 > Debian_amd64`, also removing duplicated entries.

Finally, we are expanding the feature coverage for SLES15, running all fleet test suites on those boxes.

## Why is it important?
Expand OS/Arch coverage

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

- Closes #2033

### CODE REVIEW

**Review by jlind23 — APPROVED** [review]

File: .ci/ansible/tasks/setup_test_script.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@
     - "export FLEET_URL=http://{{stackRunner}}:8220"
     - "export SKIP_PULL=1"
 
-  when: "'fleet_arm64' == nodeLabel or 'fleet_amd64' == nodeLabel or 'fleet_sles15' == nodeLabel"
+  when: "'fleet' == suite"
```
Comment (adam-stokes) [inline .ci/ansible/tasks/setup_test_script.yml:19]: much nicer thanks!

**Review by adam-stokes — APPROVED** [review]

File: .ci/ansible/playbook.yml [hunk 2]
```diff
@@ -42,7 +42,7 @@
         golang_version: 1.17
   vars:
     ansible_user: "{{nodeUser}}"
-    ansible_python_interpreter: /usr/bin/python3
+    ansible_python_interpreter: "{{ '/usr/libexec/platform-python' if ansible_distribution == 'CentOS' else '/usr/bin/python3' }}"
```
Comment (mdelapenya) [inline .ci/ansible/playbook.yml:45]: Neat!

**Comment by adam-stokes** [comment]
CentOS support is there now, we just need to resolve the artifact version missing

**Comment by mdelapenya** [comment]
> CentOS support is there now, we just need to resolve the artifact version missing

We can merge and resolve the bump in another PR

File: .ci/ansible/files/user-data-centos.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+yum install -y python3
```
Comment (mdelapenya) [inline .ci/ansible/files/user-data-centos.sh:3]: I think this file is a left over for user_data. We can remove it

Comment (mdelapenya) [reply .ci/ansible/files/user-data-centos.sh]: Removed here: https://github.com/elastic/e2e-testing/pull/2034/commits/730c89ac4a847fcff19c5cdaa6a38ca11d7ea822

**Pull request merged**